woocommerce/payment-gateway/class-sv-wc-payment-gateway.php

```diff
@@ -4382,14 +4386,41 @@ public function is_echeck_gateway() {
 	/**
 	 * Returns true if currently processing payment from block-based checkout.
 	 *
-	 * @since 5.12.0
+	 * @since 5.12.1-dev.1
+	 *
+	 * @return bool
+	 */
+	protected function is_processing_block_checkout(): bool {
+
+		return $this->is_processing_context( self::PROCESSING_CONTEXT_BLOCK ) && Blocks_Handler::is_checkout_block_in_use();
+	}
+
+
+	/**
+	 * Gets the current payment processing context.
+	 *
+	 * @since 5.12.1-dev.1
+	 * @see Gateway_Checkout_Block_Integration::prepare_payment_data()
+	 *
+	 * @return string
+	 */
+	protected function get_processing_context(  ): ?string {
+
+		return $_POST[ 'wc-' . $this->get_id_dasherized() . '-context' ] ?: null;
+	}
+
+
+	/**
+	 * Checks if we're currently in teh given payment processing context.
+	 *
+	 * @since 5.12.1-dev.1
 	 * @see Gateway_Checkout_Block_Integration::prepare_payment_data()
 	 *
 	 * @return bool
 	 */
-	protected function is_block_checkout(): bool {
```

Comment:
just concerned if we are accessing this method from any of the gateways we've been releasing so far

Reply:
Cybersource should be the only one, and it hasn't been released.

Reply:
alright then, latest commit looks good too